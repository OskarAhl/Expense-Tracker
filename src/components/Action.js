import React from 'react';

// Why stateless functional components ?
// performance - less 'overhead' to compile vs Class based component (e.g. no lifecyclehooks or state functionality)
const Action = (props) => (
    <div>
        <button
            className="big-button" 
            disabled={!props.hasOptions} 
            onClick={props.handlePick}
        >
            What should I do?
        </button>  
    </div>
);

export default Action;