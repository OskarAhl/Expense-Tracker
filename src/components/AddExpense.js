import React from 'react';

const AddExpensePage = () => (
    <div>
        Addpage
    </div>
);

export default AddExpensePage;