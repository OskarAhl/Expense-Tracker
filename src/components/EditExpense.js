import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
    return (
        // to populate expenseForm we need to pass down props.expense 
        <div>
            <ExpenseForm 
                expense={props.expense}
                onSubmit={(expense) => { 
                    props.dispatch(editExpense(props.expense.id, expense));
                    props.history.push('/');
                }}
            />
            <button onClick={ () => {
                props.dispatch(removeExpense({id: props.expense.id}))
                props.history.push('/');
            }}>
            Remove</button>
        </div>
    );
}

// HCO
const mapStateToProps = (state, props) => {
    // find expense that matches params id - and 'add to props', i.e. accessible in EditExpensePage
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    }
};

export default connect(mapStateToProps)(EditExpensePage);