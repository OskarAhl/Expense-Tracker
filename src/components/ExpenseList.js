import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// when state changes component will be re-rendered with the new state
// export unconnected (from state) component for testing
export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? 
            ( <p>No expenses</p> ) : 
            (
                props.expenses.map((expense) => (
                    <ExpenseListItem 
                        key={expense.id} 
                        {...expense}
                    />
                ))
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    // what information from store - component should be allowed to access
    return {
        expenses: selectExpenses(state.expenses, state.filters),
    };
}

export default connect(mapStateToProps)(ExpenseList);
