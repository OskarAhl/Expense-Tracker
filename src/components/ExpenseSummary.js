import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import { getExpensesTotal, getCountTotal } from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => (
    <div> 
        <h4>
            Viewing { getCountTotal(props.expenses) + ' ' } 
            totalling { numeral(getExpensesTotal(props.expenses)).format('$0,0.00') } 
        </h4>
    </div>
);

const mapStateToProps = (state) => ({
        expenses: selectExpenses(state.expenses, state.filters),
});

export default connect(mapStateToProps)(ExpenseSummary);