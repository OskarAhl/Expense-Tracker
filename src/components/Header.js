import React from 'react';

const Header = (props) => (
    <div className="header">
        <div className="container">
            <h1 className="header__title">{props.title}</h1>
            {props.subTitle && <h2 className="header__subtitle">{props.subTitle}</h2>}
        </div>
    </div>
);
// default for header - but can be changed by setting title={...} in parent. 
Header.defaultProps = {
    title: 'Indecision App'
};

export default Header;