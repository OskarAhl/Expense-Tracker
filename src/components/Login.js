import React from 'react';

const Login = () => (
    <div> 
        <p>Login Here</p>
        <button>Login</button>
    </div>
);

export default Login;
