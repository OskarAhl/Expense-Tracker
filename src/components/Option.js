import React from 'react';

const Option = (props) => (
    <div className="option">
    <p className="option__text">{props.count}. {props.optionText}</p>  
        <button 
            className="button button--link"
            onClick={(e) => {
                props.handleDeleteOption(props.optionText)
            }}
        >
        Remove
        </button>
    </div>
);
// stateless functional component
// const User = (props) => {
//     return (
//         <div>
//             <p>Name: {props.name}</p>
//             <p>Age: {props.age}</p>
//         </div>
//     )
// }
export default Option;