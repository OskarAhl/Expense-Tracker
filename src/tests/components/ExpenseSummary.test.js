import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render ExpenseSummary', () => {
    // const wrapper = shallow(<ExpenseSummary />);
    // expect(wrapper).toMatchSnapshot();
});