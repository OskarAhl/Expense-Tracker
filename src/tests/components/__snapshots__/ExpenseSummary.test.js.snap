// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary 1`] = `
<div>
  <h4>
    Viewing 
    0 expenses 
    totalling 
    $0.00
  </h4>
</div>
`;
